### PR TITLE
[feat] Gradient accumulation

### PR DESCRIPTION
**Pull request by pufanyi**
Resolved https://github.com/EvolvingLMMs-Lab/lmms-engine/issues/87

1. Add FSDP2 gradient accumulation implementation;
2. Updated the multi-node documentation.

### CODE REVIEW

**Review by kcz358 — APPROVED** [review]
LGTM for me. Not sure about the reduce scatter case for this. I think this is necessary for the fsdp module to work? Because it is hooked up for the nn.Module. Not sure about this.

If possible, please add a test case for qwenvl for gradient accumulation and make sure the other test cases for qwenvl also pass. Then I will merge this PR. Thanks!

**Comment by pufanyi** [comment]
OK! I will add a test case. For reduce scatter, I'm not very sure we can accumulate here:

image

So that we can save some reduce scatter steps.
